### PR TITLE
chore: add JSDoc comments to address.ts

### PR DESCRIPTION
**Pull request by mbokinala**

### CODE REVIEW

**Review by import-brain — DISMISSED** [review]

**Review by Shinigami92 — DISMISSED** [review]

**Review by ST-DDT — DISMISSED** [review]
Looks good to me.
However, I would suggest using consistent `.` at the end of the sentences.

**Comment by mbokinala** [comment]
> Looks good to me. However, I would suggest using consistent `.` at the end of the sentences.

updated.

File: src/address.ts [hunk 1]
```diff
@@ -382,10 +363,9 @@ export class Address {
   }
 
   /**
-   * ordinal direction
+   * Generate a random ordinal direction (northwest, southeast, etc).
```
Comment (ST-DDT) [inline src/address.ts:366]:
```suggestion
   * Returns a random ordinal direction (northwest, southeast, etc).
```

File: src/address.ts [hunk 2]
```diff
@@ -174,10 +168,9 @@ export class Address {
   //
 
   /**
-   * Returns a random localized street address
+   * Returns a random localized street address.
```
Comment (ST-DDT) [inline src/address.ts:171]:
```suggestion
   * Generates a random localized street address.
```
(not sure about this one)

File: src/address.ts [hunk 3]
```diff
@@ -147,9 +142,8 @@ export class Address {
   }
 
   /**
-   * Returns a random localized street name
+   * Returns a random localized street name.
```
Comment (ST-DDT) [inline src/address.ts:145]:
```suggestion
   * Generates a random localized street name.
```
(not sure about this one)

**Review by ST-DDT — COMMENTED** [review]
There is one minor typo.

Maybe the jsdocs should reflect the implementation of the method, what do you think?

- `Returns a` -> Select an existing element randomly
- `Generates a` -> Actually generate something new
- If both apply, `Generates a` wins.

Before you change all these, maybe you should wait for feedback from the project maintainers (which I am not).

**Review by Shinigami92 — APPROVED** [review]

File: src/address.ts [hunk 4]
```diff
@@ -365,10 +347,9 @@ export class Address {
   }
 
   /**
-   * cardinal direction
+   * Generate a random cardinal direction (north, east, south, west).
```
[the remark this answers is not in the document]
Comment (mbokinala) [reply src/address.ts:350]: This makes sense, I was initially following what was already written in the schema definitions. I think the distinction between `generates` and `returns` would make it easier to understand what the function can do. I will make this change once we hear from maintainers.

File: src/address.ts (hunk 3 above)
Comment (mbokinala) [reply src/address.ts:145]: @Shinigami92 do you know if the street names are chosen from a predetermined list?

Comment (ST-DDT) [reply src/address.ts:145]: Street names are easily generate-able. 

- `[number]th-avenue` // 5th avenue
- `[number]th-[cardinal-direction]-street` // 5th east street
- `[first_name] street` // john street
- `[adjective] [word]street` // beautiful pinestreet
- `[adjective]-street` // nice-street

We could have a list of hardcoded street names in the background too, but that shouldn't be relevant to the user.

**Comment by ST-DDT** [comment]
@mbokinala Please rebase your PR (to fix the tests) and apply the suggestions.

File: src/address.ts [hunk 5]
```diff
@@ -114,9 +112,8 @@ export class Address {
   }
 
   /**
-   * Return a random localized city prefix
+   * Returns a random localized city prefix.
```
Comment (pkuczynski) [inline src/address.ts:115]: This is missing `@example`

**Comment by ST-DDT** [comment]
@mbokinala Do you wish to continue with this or shall we fix the remaining issues?

**Comment by ST-DDT** [comment]
I think I will push this forward.

**Review by ST-DDT — APPROVED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Pull request merged**